includes/create-theme/theme-locale.php

```diff
@@ -28,6 +28,72 @@ private static function escape_text_content( $string ) {
 
 		$string = addcslashes( $string, "'" );
 
+		// Process the string to avoid escaping inner HTML markup.
+		$p = new WP_HTML_Tag_Processor( $string );
+
+		$text   = '';
+		$tokens = array();
+		while ( $p->next_token() ) {
+			$token_type    = $p->get_token_type();
+			$token_name    = strtolower( $p->get_token_name() );
+			$is_tag_closer = $p->is_tag_closer();
+
+			if ( '#tag' === $token_type ) {
+				// Add a placeholder for the token.
+				$text .= '%s';
+				if ( $is_tag_closer ) {
+					$tokens[] = "</{$token_name}>";
+				} else {
+					// Depending on the HTML tag, we may need to process attributes so they are correctly added to the placeholder.
+					switch ( $token_name ) {
+						// Handle links.
+						case 'a':
+							$href     = esc_url( $p->get_attribute( 'href' ) );
+							$target   = empty( esc_attr( $p->get_attribute( 'target' ) ) ) ? '' : ' target="_blank"';
+							$rel      = empty( esc_attr( $p->get_attribute( 'rel' ) ) ) ? '' : ' rel="nofollow noopener noreferrer"';
+							$tokens[] = "<a href=\"{$href}\"{$target}{$rel}>";
+							break;
+						// Handle inline images.
+						case 'img':
+							$src   = esc_url( $p->get_attribute( 'src' ) );
+							$style = esc_attr( $p->get_attribute( 'style' ) );
+							$alt   = esc_attr( $p->get_attribute( 'alt' ) );
+
+							CBT_Theme_Media::add_media_to_local( array( $src ) );
+							$relative_src = CBT_Theme_Media::get_media_folder_path_from_url( $src ) . basename( $src );
+							$tokens[]     = "<img style=\"{$style}\" src=\"' . esc_url( get_stylesheet_directory_uri() ) . '{$relative_src}\" alt=\"{$alt}\">";
+							break;
+						// Handle highlights.
+						case 'mark':
+							$style    = esc_attr( $p->get_attribute( 'style' ) );
+							$class    = esc_attr( $p->get_attribute( 'class' ) );
+							$tokens[] = "<mark style=\"{$style}\" class=\"{$class}\">";
+							break;
+						// Otherwise, just add the tag opener.
+						default:
+							$tokens[] = "<{$token_name}>";
+							break;
+					}
+				}
+			} else {
+				// If it's not a tag, just add the text content.
+				$text .= esc_html( $p->get_modifiable_text() );
+			}
+		}
+		// If tokens is not empty, format the string using sprintf.
+		if ( ! empty( $tokens ) ) {
+			// Format the string, replacing the placeholders with the formatted tokens.
+			return "<?php /* Translators: %s are html tags */ echo sprintf( esc_html__( '$text', '" . wp_get_theme()->get( 'TextDomain' ) . "' ), " . implode(
```

Comment:
`Translators: %s are html tags` doesn't seem like a very helpful comment. From what I understand, the best practices for using multiple placeholders includes

- Use numbered placeholders (`%1$s` rather than `%s`)
- Name the tag in the comment: (e.g. `/* translators: %1$s: start of link (<a>), %2$s: end of link (</a>) */`)

Reply:
If the content itself is having a `%s`, it breaks the page because of the improper closings.
Replacing such instances with `%%s` or `%%1$s` will fix things.